packages/flutter/lib/src/widgets/scrollable_helpers.dart

```diff
@@ -460,17 +460,17 @@ class ScrollAction extends ContextAction<ScrollIntent> {
         return true;
       }());
 
-      if (primaryScrollController.position.context.notificationContext == null
-          && Scrollable.maybeOf(primaryScrollController.position.context.notificationContext!) == null) {
+      if (primaryScrollController.position.context.notificationContext != null) {
+        state = Scrollable.maybeOf(primaryScrollController.position.context.notificationContext!);
+      }
+      if (state == null) {
         return;
       }
-      state = Scrollable.maybeOf(primaryScrollController.position.context.notificationContext!);
     }
-    assert(state != null, '$ScrollAction was invoked on a context that has no scrollable parent');
```

Comment:
I agree it's a bit strange for this method to use a combination of `assert`s and early `return`s, rather than sticking with one or the other.

Reply:
Yeah, this assert was also marked as unnecessary by the linter after applying the patch.